sw drive

mirapanigrahy opened the pull request.

Commit:
moved swerve drive code to new repo

Commit:
fixed import errors in 2024 update

Commit:
removed out of date sim spark json file

Commit:
added drive system to auto handler

Commit:
added update auto method to drive fsm

Commit:
cleaned up teleop input file

AknA13 committed:
Pose resets to (0,0,0) at start of auto and teleop

AknA13 committed:
fixed update(null) nullpointer issue

AknA13 committed:
removed stale comments

Commit:
fixed bugs with running chassis

Commit:
Merge branch 'SW-Drive' of https://github.com/Tino-FRC-2473/FRC2024 into SW-Drive

Commit:
resolved merge conflicts

AknA13 committed:
working movement along multi-point paths

AknA13 committed:
Merge branch 'test/week-1/SW-Drive/point_to_point_driving' into SW-Drive

AknA13 committed:
removed unused travel angle

AknA13 committed:
checkstyle

Commit:
added seperate tag and object alignment states

Commit:
added auto path chooser for choosing paths via shuffleboard

Commit:
checkstyle

AG6GR commented on vendordeps/navx_frc.json:
```diff
@@ -0,0 +1,35 @@
+    "mavenUrls": [
+        "https://repo1.maven.org/maven2/"
+    ],
+    "jsonUrl": "https://www.kauailabs.com/dist/frc/2022/navx_frc.json",
```
This is the 2022 version of the NavX library, which is incompatible with the latest WPILib. The 2024 library is already included under vendordeps.

AG6GR commented on src/main/java/frc/robot/AutoPathChooser.java:
```diff
@@ -0,0 +1,63 @@
+	 */
+	public AutoPathChooser() {
+		autoPathChooser = new SendableChooser<>();
+		autoPathChooser.setDefaultOption("Path 1", AutoPath.PATH1);
```
Are there more descriptive names we could give the paths?

AG6GR commented on src/main/java/frc/robot/systems/MAXSwerveModule.java:
Can you put a comment in the README.md with the link to the repo you got this code from? We should cite the original source. Plus it will make it easier for people reading the code for the first time.

Commit:
deleted old naxx file

Commit:
added new naxX file

Commit:
fixing bugs with auto path chooser

Commit:
test auto path chooser -- working

Commit:
added maxswerve module copywright and code link

Commit:
added more descriptive names for the paths on the path chooser

AG6GR committed:
Update build job to Java 17

Fix build issue due to out of data JDK.

Commit:
implemented blue aliance auto paths

koalabearrawr committed:
added red alliance auto path temp code

added the code for the auto paths assuming the robot is on the red alliance. some angles are to be determined.

Commit:
reset odometry when enabled

Commit:
Merge branch 'SW-Drive' of https://github.com/Tino-FRC-2473/FRC2024 into SW-Drive

Commit:
commented out 0 for start auto

Commit:
added PENDING state in AutoHandlerSystem and declared constants for auto path coordinates in SwerveConstants

Commit:
paths 1 and 4 tested -- need to figure out accelation constant

Commit:
paths 1 and 4 tested -- need to figure out accelation constant

Commit:
Merge branch 'SW-Drive' of https://github.com/Tino-FRC-2473/FRC2024 into SW-Drive

Commit:
ensured that the acceleration constant would always produce a speed within max speed

Commit:
tested path 2, added turn movement for paths

Commit:
checkstyle and fixed points

Commit:
cleaned up code by removing uncessary comments, constants, and prints

Commit:
checkstyle

mirapanigrahy committed:
Merge branch 'main' into SW-Drive

Commit:
checkstyle + merge cleanup

Commit:
added pause(seconds) for the pending state in drive -- to be implemented

AknA13 commented on README.md:
```diff
@@ -32,3 +32,31 @@ Everyone should read and follow the rules in "[How to write a Git Commit Message
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Do we need this text in our README

mirapanigrahy replied:
yes sunny asked to add it

AknA13 commented on src/main/java/frc/robot/HardwareMap.java:
```diff
@@ -1,33 +1,27 @@
 	public static final int CAN_ID_SPARK_DRIVE_BACK_RIGHT = 2;
 	public static final int CAN_ID_SPARK_DRIVE_FRONT_LEFT = 3;
 	public static final int CAN_ID_SPARK_DRIVE_BACK_LEFT = 4;
 
```
Delete old CAN IDs

AknA13 commented on src/main/java/frc/robot/RaspberryPI.java:
```diff
@@ -0,0 +1,98 @@
+	 * @param id takes the id of tag
+	 * @return the yaw to the tag
+	 */
+	public double getAprilTagYaw(int id) {
```
The method names are switched, this should be getAprilTagRoll(int id)

AknA13 commented on src/main/java/frc/robot/RaspberryPI.java:
```diff
@@ -0,0 +1,98 @@
+	 * @param id takes the id of tag
+	 * @return the pitch to the tag
+	 */
+	public double getAprilTagPitch(int id) {
```
The method names are switched, this should be getAprilTagYaw(int id)

AknA13 commented on src/main/java/frc/robot/RaspberryPI.java:
```diff
@@ -0,0 +1,98 @@
+	 * @param id takes the id of tag
+	 * @return the roll to the tag
+	 */
+	public double getAprilTagRoll(int id) {
```
The method names are switched, this should be getAprilTagPitch(int id)

AknA13 commented on src/main/java/frc/robot/systems/DriveFSMSystem.java:
```diff
@@ -0,0 +1,693 @@
+				// 	|| ydist == AutoConstants.UNABLE_TO_SEE_TAG_CONSTANT);
+				// SmartDashboard.putBoolean("Can See Tag", canSee);
+				// System.out.println("x tag: " + rpi.getAprilTagX(1));
+				// driveToTag(0, xdist, 0);
```
For now, just remove all these comments for both the CV states. I'll add the tested code later in the Automation PR

AknA13 commented on src/main/java/frc/robot/systems/DriveFSMSystem.java:
```diff
@@ -0,0 +1,693 @@
+	 */
+	public boolean driveToPose(Pose2d pose) {
+		double x = pose.getX() / 2; // remove / 2 after testing
+		double y = pose.getY() / 2; // remove / 2 after testing
```
Remove the /2 for x and y

AknA13 commented on src/main/java/frc/robot/systems/DriveFSMSystem.java:
```diff
@@ -0,0 +1,693 @@
+	 * @param seconds seconds to wait
+	 * @return whether or not the wait time has been completed
+	 */
+	public boolean pause(double seconds) {
```
Implement this

AknA13 commented on src/main/java/frc/robot/systems/DriveFSMSystem.java:
```diff
@@ -0,0 +1,693 @@
+	 * @param yDist constantly updating y distance to the point
+	 * @param rotFinal constantly updating angular difference to the point
+	 */
+	public void driveToTag(double xDist, double yDist, double rotFinal) {
```
Delete this method for now, will add back in the automation PR

mirapanigrahy replied:
Thats so sad, it will be much easier to merge automation speaker alignment code into this pr bc they're already pretty close. I've been merging automation updates into drive. Alr its okay ill remove it from drive for the time being.

AknA13 commented on src/main/java/frc/robot/systems/DriveFSMSystem.java:
```diff
@@ -0,0 +1,693 @@
+	 * @param dist constantly updating distance to the object
+	 * @param rotFinal constantly updating angular difference to the point
+	 */
+	public void driveUntilObject(double dist, double rotFinal) {
```
Delete this method for now, will add back in the automation PR

mirapanigrahy committed:
Test/week 1/apriltags and optimization (#4)

* added rpi java code

* added CV python files

* added apriltag module and test script

* streamlined code for only apriltag computation, replaced cv2.aruco with apriltag library

* april tag code

* working calibration code

* added apriltag network table code

* added pupil apriltag code

* added working aruco code on opencv 4.5

* added calibration data for cam 1

* code reformat

* removed image annotation to speed up april tag pipeline

* modified driveToTag() to only align horizontally for testing purposes

* tested april tag alignment with robot

* switched april tag pose from camera-relative to tag-relative

* switched aruco detection method to opencv 4.5

* added only rotation in driveToTag

* added transformation to april tag values for alignment

* negated x speed

* saved last seen value of april tag info so it can be used for alignment

* added apriltag yaw

* made robot turn to april tag instead of parallel

* added tag values on shuffleboard

* changed constants

* commented out transformations

* only y movement

* removed abs val from april tag vals

* fixing merge conflicts

* fixed merge errors

---------

Co-authored-by: Akshay Anand <95551900+AknA13@users.noreply.github.com>
Co-authored-by: Tanmayi <tanmayi.dasari@gmail.com>
Co-authored-by: AknA13 <akshaya91306@gmail.com>

Commit:
fixing errors, removing uncessary code

Commit:
removed untested automation code

Commit:
cleared automation code

Commit:
added constant

Commit:
added constant

Commit:
implemented pause method
